Client/Frontend/Browser/Playlist/Cache/PlaylistCacheLoader.swift

```diff
@@ -0,0 +1,277 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AVFoundation
+import WebKit
+import MobileCoreServices
+import YubiKit
+
+public class PlaylistMimeTypeDetector {
+    private(set) var mimeType: String = ""
+    private(set) var fileExtension: String = ""
+    
+    init(mimeType: String) {
+        self.mimeType = mimeType
+        
+        let mimeTypeMap = [
+            "video/webm": "webm",
+            "application/ogg": "ogg",
+            "audio/x-wav": "wav",
+            "video/x-msvideo": "avi",
+            "audio/mpeg": "mp4",
+            "audio/flac": "flac",
+            "video/mp4": "mp4",
+            "application/x-mpegURL": "mpg",
+            "application/vnd.apple.mpegurl": "mpg"
+        ]
+        
+        for (key, value) in mimeTypeMap {
+            if mimeType.contains(key) {
+                self.fileExtension = value
+                break
+            }
+        }
```

Comment:
Is there a reason you aren't doing something like:

```suggestion
        self.fileExtension = mimeTypeMap[mimeType]
```